### DISCUSSION TITLE
Pytest 8.3.3 breaks fixture registration for class based plugins when fixtures are dynamically added to the instance only

### DISCUSSION THREAD

**Discussion by lovetheguitar** [pytest-dev/pytest#12851] [General]
In c6a529032231ccddd3040e8ab1a5a756eb9ea4a0 the detection of fixtures was changed in a way that broke our dynamic fixture registration. Before fixtures were looked up on the instance, but now they will be looked up on the class of a plugin instance. 
[Relevant line](https://github.com/pytest-dev/pytest/commit/c6a529032231ccddd3040e8ab1a5a756eb9ea4a0#diff-49027cfd80e14edac9b0fae71f7228a408a09599f66a7815839ce8c3ae2ab84fR1730).

Thus I'm wondering what the officially supported `plugin`s one may feed into https://github.com/pytest-dev/pytest/blob/7add2af7087dc32af19a38f5950a5b64a19d1de5/src/_pytest/config/__init__.py#L489 are?

Reproduction:
**test_demo.py**
```py
"""Succeeds for <8.3.3, but fails since."""
import pytest


pytest_plugins = ["pytester"]


def test_demo_fixture_present_on_plugin_instance_only_are_not_registered_anymore(pytester: pytest.Pytester) -> None:
    pytester.makeconftest(
        """
        from typing import Callable
        import types
        import pytest
        
        class PluginContainer:
            def add_fixture(
                self, fixture_function: Callable[..., object], fixture_name: str
            ) -> None:
                '''Add pytest fixture function with specified name to Plugin Container.
        
                .. note:: For pytest to find the fixture its apparently important that methods are
                    bound to the instance!
                :param fixture_function: a pytest fixture object
                :param fixture_name: the desired name of the fixture
                '''
                setattr(self, fixture_name, types.MethodType(fixture_function, self))
                
        def pytest_configure(config: pytest.Config) -> None:
            plugin_container = PluginContainer()
            
            @pytest.fixture
            def demo_fixture(self, request):
                ...
                
            plugin_container.add_fixture(demo_fixture, "fixture_name")
    
            config.pluginmanager.register(plugin_container, __name__)
    """)


    result = pytester.runpytest("--fixtures")
    assert result.ret == pytest.ExitCode.OK
    result.stdout.fnmatch_lines(["*fixture_name*"])
```

**Comment by RonnyPfannschmidt** [comment]
looks like a "bug" as we never explicitly supported dropping dynamicfixtures into a object instances
we should look into `__dict__` aswell when applicable

**Reply by RonnyPfannschmidt** [reply]
i think in this particular case we may use inspect.getattr_static

**Reply by lovetheguitar** [reply]
I tried playing around with `inspect.getattr_static`, but had lots of failing tests.

Since this issue doesn’t seem to affect others, I refactored the code to move `PluginContainer` class creation into a method. This method is then called to create individual classes for each plugin registration.

> looks like a "bug" as we never explicitly supported dropping dynamicfixtures into a object instances
Is there any documentation what actually is explicitly supported?
